x-pack/plugins/apm/public/components/app/correlations/use_failed_transactions_correlations.ts

```diff
@@ -0,0 +1,257 @@
+  );
+  const setResponse = useMemo(
+    () => debounce(setResponseUnDebounced, DEBOUNCE_INTERVAL),
+    []
+  );
+
+  const abortCtrl = useRef(new AbortController());
+
+  const startFetch = useCallback(async () => {
+    abortCtrl.current.abort();
+    abortCtrl.current = new AbortController();
+
+    setResponse({
+      ...getInitialResponse(),
+      isRunning: true,
+      // explicitly set these to undefined to override a possible previous state.
+      error: undefined,
+      failedTransactionsCorrelations: undefined,
+      percentileThresholdValue: undefined,
+      overallHistogram: undefined,
+      errorHistogram: undefined,
+      fieldStats: undefined,
+    });
+    setResponse.flush();
+
+    try {
+      // `responseUpdate` will be enriched with additional data with subsequent
+      // calls to the overall histogram, field candidates, field value pairs, correlation results
+      // and histogram data for statistically significant results.
+      const responseUpdate: FailedTransactionsCorrelationsResponse = {
+        ccsWarning: false,
+      };
+
+      const [overallHistogramResponse, errorHistogramRespone] =
+        await Promise.all([
+          // Initial call to fetch the overall distribution for the log-log plot.
+          callApmApi({
+            endpoint: 'POST /internal/apm/latency/overall_distribution',
+            signal: abortCtrl.current.signal,
+            params: {
+              body: {
+                ...fetchParams,
+                percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+              },
+            },
+          }),
+          callApmApi({
+            endpoint: 'POST /internal/apm/latency/overall_distribution',
+            signal: abortCtrl.current.signal,
+            params: {
+              body: {
+                ...fetchParams,
+                percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+                termFilters: [
+                  {
+                    fieldName: EVENT_OUTCOME,
+                    fieldValue: EventOutcome.failure,
+                  },
+                ],
+              },
+            },
+          }),
+        ]);
+
+      const { overallHistogram, percentileThresholdValue } =
+        overallHistogramResponse;
+      const { overallHistogram: errorHistogram } = errorHistogramRespone;
+
+      responseUpdate.errorHistogram = errorHistogram;
+      responseUpdate.overallHistogram = overallHistogram;
+      responseUpdate.percentileThresholdValue = percentileThresholdValue;
+
+      if (abortCtrl.current.signal.aborted) {
+        return;
+      }
+
+      setResponse({
+        ...responseUpdate,
+        loaded: LOADED_OVERALL_HISTOGRAM,
+      });
+      setResponse.flush();
+
+      const { fieldCandidates: candidates } = await callApmApi({
+        endpoint: 'GET /internal/apm/correlations/field_candidates',
+        signal: abortCtrl.current.signal,
+        params: {
+          query: fetchParams,
+        },
+      });
```

Comment:
This can be parallelised as well no?

Reply:
I parallelised the first two requests to get the chart data first so we can show the chart as soon as it's available. This one is the first of the requests of the analysis, the requests after that one depends on its output.

Reply:
I'm not sure if I fully understand, but there's no need (AFAICT) to delay starting the request until the data for the chart has been fully loaded. E.g. you can start the request but only await it once the data for the charts have loaded.